File: Nx3CS/bin/test/src/nx3/PSystemBar.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PSystemBar : global::haxe.lang.HxObject {
		public    PSystemBar(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PSystemBar(global::nx3.PBar bar, global::nx3.PBarConfig barConfig, global::nx3.PSystembarMeasurements barMeasurements, object actAttributes, object caAttributes){
			unchecked {
				#line 23 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				global::nx3.PSystemBar.__hx_ctor_nx3_PSystemBar(this, bar, barConfig, barMeasurements, actAttributes, caAttributes);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PSystemBar(global::nx3.PSystemBar __temp_me122, global::nx3.PBar bar, global::nx3.PBarConfig barConfig, global::nx3.PSystembarMeasurements barMeasurements, object actAttributes, object caAttributes){
			unchecked {
				#line 39 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				__temp_me122.stretchamount = ((double) (0) );
				#line 25 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				__temp_me122.bar = bar;
				__temp_me122.bar.systembar = __temp_me122;
				__temp_me122.barConfig = barConfig;
				__temp_me122.barMeasurements = barMeasurements;
				__temp_me122.actAttributes = actAttributes;
				__temp_me122.caAttributes = caAttributes;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				return new global::nx3.PSystemBar(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				return new global::nx3.PSystemBar(((global::nx3.PBar) (arr[0]) ), ((global::nx3.PBarConfig) (arr[1]) ), ((global::nx3.PSystembarMeasurements) (arr[2]) ), ((object) (arr[3]) ), ((object) (arr[4]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PBar bar;
		
		public  global::nx3.PBarConfig barConfig;
		
		public  object actAttributes;
		
		public  object caAttributes;
		
		public  double stretchamount;
		
		public virtual   void setBarStretch(double amount){
			unchecked {
				#line 42 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				if (( amount == this.stretchamount )) {
					#line 42 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
					return ;
				}
				
				#line 43 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				global::nx3.PBarStretchCalculator calculator = new global::nx3.PBarStretchCalculator(((global::nx3.PSystemBar) (this) ));
				if (( amount == 0 )) {
					#line 46 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
					calculator.resetStretch();
				}
				 else {
					#line 51 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
					calculator.stretch(amount);
				}
				
			}
			#line default
		}
		
		
		public  global::nx3.PSystembarMeasurements barMeasurements;
		
		public virtual   global::nx3.PSystembarMeasurements getBarMeasurements(){
			unchecked {
				#line 58 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				if (( this.barMeasurements != default(global::nx3.PSystembarMeasurements) )) {
					#line 58 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
					return this.barMeasurements;
				}
				
				#line 60 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				this.barMeasurements = new global::nx3.PSystembarMeasurements(((global::nx3.PBar) (this.bar) )).init(this.actAttributes, this.barConfig, this.caAttributes);
				return this.barMeasurements;
			}
			#line default
		}
		
		
		public  double xposition;
		
		public virtual   double getXPosition(){
			unchecked {
				#line 67 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				return this.xposition;
			}
			#line default
		}
		
		
		public override   double __hx_setField_f(string field, int hash, double @value, bool handleProperties){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				switch (hash){
					case 740546849:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.xposition = ((double) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 1313087901:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.stretchamount = ((double) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 1332786453:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.caAttributes = ((object) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 1306671305:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.actAttributes = ((object) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					default:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return base.__hx_setField_f(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				switch (hash){
					case 740546849:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.xposition = ((double) (global::haxe.lang.Runtime.toDouble(@value)) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 690070378:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.barMeasurements = ((global::nx3.PSystembarMeasurements) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 1313087901:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.stretchamount = ((double) (global::haxe.lang.Runtime.toDouble(@value)) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 1332786453:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.caAttributes = ((object) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 1306671305:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.actAttributes = ((object) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 1836915957:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.barConfig = ((global::nx3.PBarConfig) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					case 4895187:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.bar = ((global::nx3.PBar) (@value) );
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return @value;
					}
					
					
					default:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				switch (hash){
					case 355217867:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getXPosition") ), ((int) (355217867) ))) );
					}
					
					
					case 740546849:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.xposition;
					}
					
					
					case 1541882740:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getBarMeasurements") ), ((int) (1541882740) ))) );
					}
					
					
					case 690070378:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.barMeasurements;
					}
					
					
					case 1737368244:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("setBarStretch") ), ((int) (1737368244) ))) );
					}
					
					
					case 1313087901:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.stretchamount;
					}
					
					
					case 1332786453:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.caAttributes;
					}
					
					
					case 1306671305:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.actAttributes;
					}
					
					
					case 1836915957:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.barConfig;
					}
					
					
					case 4895187:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.bar;
					}
					
					
					default:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_getField_f(string field, int hash, bool throwErrors, bool handleProperties){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				switch (hash){
					case 740546849:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.xposition;
					}
					
					
					case 1313087901:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.stretchamount;
					}
					
					
					case 1332786453:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return ((double) (global::haxe.lang.Runtime.toDouble(this.caAttributes)) );
					}
					
					
					case 1306671305:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return ((double) (global::haxe.lang.Runtime.toDouble(this.actAttributes)) );
					}
					
					
					default:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return base.__hx_getField_f(field, hash, throwErrors, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				switch (hash){
					case 355217867:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.getXPosition();
					}
					
					
					case 1541882740:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return this.getBarMeasurements();
					}
					
					
					case 1737368244:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						this.setBarStretch(((double) (global::haxe.lang.Runtime.toDouble(dynargs[0])) ));
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						break;
					}
					
					
					default:
					{
						#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				return default(object);
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				baseArr.push("xposition");
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				baseArr.push("barMeasurements");
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				baseArr.push("stretchamount");
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				baseArr.push("caAttributes");
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				baseArr.push("actAttributes");
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				baseArr.push("barConfig");
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				baseArr.push("bar");
				#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
				{
					#line 21 "F:\\nx3\\src\\nx3\\PSystemBar.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/PSystemBar.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PSystemBar extends haxe.lang.HxObject
{
	public    PSystemBar(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PSystemBar(nx3.PBar bar, nx3.PBarConfig barConfig, nx3.PSystembarMeasurements barMeasurements, java.lang.Object actAttributes, java.lang.Object caAttributes)
	{
		nx3.PSystemBar.__hx_ctor_nx3_PSystemBar(this, bar, barConfig, barMeasurements, actAttributes, caAttributes);
	}
	
	
	public static   void __hx_ctor_nx3_PSystemBar(nx3.PSystemBar __temp_me115, nx3.PBar bar, nx3.PBarConfig barConfig, nx3.PSystembarMeasurements barMeasurements, java.lang.Object actAttributes, java.lang.Object caAttributes)
	{
		__temp_me115.stretchamount = ((double) (0) );
		__temp_me115.bar = bar;
		__temp_me115.bar.systembar = __temp_me115;
		__temp_me115.barConfig = barConfig;
		__temp_me115.barMeasurements = barMeasurements;
		__temp_me115.actAttributes = actAttributes;
		__temp_me115.caAttributes = caAttributes;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PSystemBar(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PSystemBar(((nx3.PBar) (arr.__get(0)) ), ((nx3.PBarConfig) (arr.__get(1)) ), ((nx3.PSystembarMeasurements) (arr.__get(2)) ), ((java.lang.Object) (arr.__get(3)) ), ((java.lang.Object) (arr.__get(4)) ));
	}
	
	
	public  nx3.PBar bar;
	
	public  nx3.PBarConfig barConfig;
	
	public  java.lang.Object actAttributes;
	
	public  java.lang.Object caAttributes;
	
	public  double stretchamount;
	
	public   void setBarStretch(double amount)
	{
		if (( amount == this.stretchamount )) 
		{
			return ;
		}
		
		nx3.PBarStretchCalculator calculator = new nx3.PBarStretchCalculator(((nx3.PSystemBar) (this) ));
		if (( amount == 0 )) 
		{
			calculator.resetStretch();
		}
		 else 
		{
			calculator.stretch(amount);
		}
		
	}
	
	
	public  nx3.PSystembarMeasurements barMeasurements;
	
	public   nx3.PSystembarMeasurements getBarMeasurements()
	{
		if (( this.barMeasurements != null )) 
		{
			return this.barMeasurements;
		}
		
		this.barMeasurements = new nx3.PSystembarMeasurements(((nx3.PBar) (this.bar) )).init(this.actAttributes, this.barConfig, this.caAttributes);
		return this.barMeasurements;
	}
	
	
	public  double xposition;
	
	public   double getXPosition()
	{
		return this.xposition;
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef592 = true;
			switch (field.hashCode())
			{
				case -1398365791:
				{
					if (field.equals("xposition")) 
					{
						__temp_executeDef592 = false;
						this.xposition = ((double) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 1610149769:
				{
					if (field.equals("actAttributes")) 
					{
						__temp_executeDef592 = false;
						this.actAttributes = ((java.lang.Object) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -729473123:
				{
					if (field.equals("stretchamount")) 
					{
						__temp_executeDef592 = false;
						this.stretchamount = ((double) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -212192171:
				{
					if (field.equals("caAttributes")) 
					{
						__temp_executeDef592 = false;
						this.caAttributes = ((java.lang.Object) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef592) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef593 = true;
			switch (field.hashCode())
			{
				case -1398365791:
				{
					if (field.equals("xposition")) 
					{
						__temp_executeDef593 = false;
						this.xposition = ((double) (haxe.lang.Runtime.toDouble(value)) );
						return value;
					}
					
					break;
				}
				
				
				case 97299:
				{
					if (field.equals("bar")) 
					{
						__temp_executeDef593 = false;
						this.bar = ((nx3.PBar) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 630515690:
				{
					if (field.equals("barMeasurements")) 
					{
						__temp_executeDef593 = false;
						this.barMeasurements = ((nx3.PSystembarMeasurements) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 632246581:
				{
					if (field.equals("barConfig")) 
					{
						__temp_executeDef593 = false;
						this.barConfig = ((nx3.PBarConfig) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -729473123:
				{
					if (field.equals("stretchamount")) 
					{
						__temp_executeDef593 = false;
						this.stretchamount = ((double) (haxe.lang.Runtime.toDouble(value)) );
						return value;
					}
					
					break;
				}
				
				
				case 1610149769:
				{
					if (field.equals("actAttributes")) 
					{
						__temp_executeDef593 = false;
						this.actAttributes = ((java.lang.Object) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -212192171:
				{
					if (field.equals("caAttributes")) 
					{
						__temp_executeDef593 = false;
						this.caAttributes = ((java.lang.Object) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef593) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef594 = true;
			switch (field.hashCode())
			{
				case -1343326453:
				{
					if (field.equals("getXPosition")) 
					{
						__temp_executeDef594 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getXPosition"))) );
					}
					
					break;
				}
				
				
				case 97299:
				{
					if (field.equals("bar")) 
					{
						__temp_executeDef594 = false;
						return this.bar;
					}
					
					break;
				}
				
				
				case -1398365791:
				{
					if (field.equals("xposition")) 
					{
						__temp_executeDef594 = false;
						return this.xposition;
					}
					
					break;
				}
				
				
				case 632246581:
				{
					if (field.equals("barConfig")) 
					{
						__temp_executeDef594 = false;
						return this.barConfig;
					}
					
					break;
				}
				
				
				case 167164340:
				{
					if (field.equals("getBarMeasurements")) 
					{
						__temp_executeDef594 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getBarMeasurements"))) );
					}
					
					break;
				}
				
				
				case 1610149769:
				{
					if (field.equals("actAttributes")) 
					{
						__temp_executeDef594 = false;
						return this.actAttributes;
					}
					
					break;
				}
				
				
				case 630515690:
				{
					if (field.equals("barMeasurements")) 
					{
						__temp_executeDef594 = false;
						return this.barMeasurements;
					}
					
					break;
				}
				
				
				case -212192171:
				{
					if (field.equals("caAttributes")) 
					{
						__temp_executeDef594 = false;
						return this.caAttributes;
					}
					
					break;
				}
				
				
				case 63167860:
				{
					if (field.equals("setBarStretch")) 
					{
						__temp_executeDef594 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("setBarStretch"))) );
					}
					
					break;
				}
				
				
				case -729473123:
				{
					if (field.equals("stretchamount")) 
					{
						__temp_executeDef594 = false;
						return this.stretchamount;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef594) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef595 = true;
			switch (field.hashCode())
			{
				case -1398365791:
				{
					if (field.equals("xposition")) 
					{
						__temp_executeDef595 = false;
						return this.xposition;
					}
					
					break;
				}
				
				
				case 1610149769:
				{
					if (field.equals("actAttributes")) 
					{
						__temp_executeDef595 = false;
						return ((double) (haxe.lang.Runtime.toDouble(this.actAttributes)) );
					}
					
					break;
				}
				
				
				case -729473123:
				{
					if (field.equals("stretchamount")) 
					{
						__temp_executeDef595 = false;
						return this.stretchamount;
					}
					
					break;
				}
				
				
				case -212192171:
				{
					if (field.equals("caAttributes")) 
					{
						__temp_executeDef595 = false;
						return ((double) (haxe.lang.Runtime.toDouble(this.caAttributes)) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef595) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef596 = true;
			switch (field.hashCode())
			{
				case -1343326453:
				{
					if (field.equals("getXPosition")) 
					{
						__temp_executeDef596 = false;
						return this.getXPosition();
					}
					
					break;
				}
				
				
				case 63167860:
				{
					if (field.equals("setBarStretch")) 
					{
						__temp_executeDef596 = false;
						this.setBarStretch(((double) (haxe.lang.Runtime.toDouble(dynargs.__get(0))) ));
					}
					
					break;
				}
				
				
				case 167164340:
				{
					if (field.equals("getBarMeasurements")) 
					{
						__temp_executeDef596 = false;
						return this.getBarMeasurements();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef596) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			
		}
		
		return null;
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("xposition");
		baseArr.push("barMeasurements");
		baseArr.push("stretchamount");
		baseArr.push("caAttributes");
		baseArr.push("actAttributes");
		baseArr.push("barConfig");
		baseArr.push("bar");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


